File: CarAuction.Backend/CarAuction.Domain/Entities/Auction.cs
namespace CarAuction.Domain.Entities;

public class Auction
{
    public int Id { get; set; }
    public int ListingId { get; set; }
    public decimal StartingPrice { get; set; }
    public decimal CurrentPrice { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public decimal MinBidIncrement { get; set; }
    public int? WinnerUserId { get; set; }
    public string Status { get; set; } = "active";

    // Navigation properties
    public Listing? Listing { get; set; }
    public User? WinnerUser { get; set; }
}

File: CarAuction.Frontend/src/types/auction.ts
export interface Auction {
  id: number;
  listingId: number;
  title?: string;
  description?: string;
  sellerId?: number;
  startingPrice: number;
  currentPrice: number;
  startTime: string;
  endTime: string;
  minBidIncrement: number;
  winnerUserId: number | null;
  status: 'active' | 'completed' | 'expired' | 'cancelled';
  imageId?: number | null;
  images?: import('./common').Image[];
}

export interface AuctionDetailResponse extends Auction {
  title: string;
  sellerId: number;
}

export interface AuctionResultDto {
  auctionId: number;
  winnerUserId: number;
  winnerName: string;
  winningPrice: number;
  status: string;
}

export interface AuctionExtendedDto {
  auctionId: number;
  newEndTime: string;
  extendedSeconds: number;
}

export interface BidDto {
  auctionId: number;
  amount: number;
  bidderId: number;
  bidderName: string;
  timestamp: string;
}
